### PR TITLE
Upgraded hyrax to the latest codebase from github master branch

### LINKED ISSUE #110: Upgrade Hyrax dependencies
### Descriptive summary

- [x] Update to latest Hyrax 2.0-alpha commit
- [x] Gemnasium: https://gemnasium.com/github.com/ucsdlib/horton/alerts

### Rationale

Addressing vulnerabilities in dependencies, stay up to date with Hyrax changes.

### PR DESCRIPTION
**Pull request by lsitu**
Fixes #110
Upgraded hyrax to the latest codebase from github master branch and fixed the errors.

Changes proposed in this pull request:
* Upgraded hyrax
* Fixed the issues with test failures.
* Refactored codes.
* Updated local pages to follow hyrax for editing.

@ucsdlib/developers - please review

### CODE REVIEW

File: app/controllers/concerns/batch_import_controller_behavior.rb [hunk 1]
```diff
@@ -24,23 +34,41 @@ def self.form_class(_ = nil)
...
+    end
+
+    def handle_payload_concern!
+      unsafe_pc = params.fetch(:batch_import_item, {})[:payload_concern]
+      # Calling constantize on user params is disfavored (per brakeman),
+      # so we sanitize by matching it against an authorized model.
+      safe_pc = Hyrax::SelectTypeListPresenter
+                .new(current_user).authorized_models.map(&:to_s).find { |x| x == unsafe_pc }
+      raise CanCan::AccessDenied, "Cannot create an object of class '#{unsafe_pc}'" unless safe_pc
+      # authorize! :create, safe_pc
```
Comment (mcritchlow) [inline app/controllers/concerns/batch_import_controller_behavior.rb:49]: Just curious as to why this commented out line is here. Was this from upstream hyrax?

File: app/actors/hyrax/actors/environment.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module Hyrax
+  module Actors
+    class Environment
```
Comment (mcritchlow) [inline app/actors/hyrax/actors/environment.rb:3]: I assume this is a new class introduced in the recent hyrax change? I'm just curious why we need a local instance? Did you customize this for our needs?

File: app/controllers/hyrax/pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+module Hyrax
+  # Shows the about and help page
+  class PagesController < ApplicationController
```
Comment (mcritchlow) [inline app/controllers/hyrax/pages_controller.rb:3]: Why do we need a local copy of this? Are we introducing customization that requires it?

File: app/schemas/general_schema.rb [hunk 4]
```diff
@@ -44,7 +44,7 @@ class GeneralSchema < ActiveTriples::Schema
   property :copyright_jurisdiction, predicate: ::RDF::Vocab::PREMIS.hasCopyrightJurisdiction
 
   # xsd:anyURI with CVs
-  property :rights_override, predicate: VisibilityService::PREDICATE_RIGHTS_OVERRIDE, multiple: false
+  # property :rights_override, predicate: VisibilityService::PREDICATE_RIGHTS_OVERRIDE, multiple: false
```
Comment (mcritchlow) [inline app/schemas/general_schema.rb:47]: Should this line just be deleted? We have in git if we need to revert for some reason.

File: config/initializers/hydra_config.rb [hunk 5]
```diff
@@ -19,4 +19,6 @@
   #
   # Specify the user model
   # config.user_model = 'User'
+
+  config.user_key_field = Devise.authentication_keys.first
```
Comment (mcritchlow) [inline config/initializers/hydra_config.rb:23]: why do we need this, or rather, what is it being used for? (just curious)

**Review by mcritchlow — APPROVED** [review]
Overall this LGTM! Thanks @lsitu 👍 

I mostly just have a couple questions, to try and understand a bit more of the context for these changes.

File: app/controllers/concerns/batch_import_controller_behavior.rb (hunk 1 above)
Comment (lsitu) [reply app/controllers/concerns/batch_import_controller_behavior.rb:49]: Yes. It's from upstream hyrax.

File: app/controllers/hyrax/pages_controller.rb (hunk 3 above)
Comment (lsitu) [reply app/controllers/hyrax/pages_controller.rb:3]: Yes. We need to customize it to add in our pages.

File: app/schemas/general_schema.rb (hunk 4 above)
Comment (lsitu) [reply app/schemas/general_schema.rb:47]: It's moved to module RightsOverrideBehavior as refactoring during fixing the issues with filesets.

File: config/initializers/hydra_config.rb (hunk 5 above)
Comment (lsitu) [reply config/initializers/hydra_config.rb:23]: Yes, It's coming from a hyrax deprecation warning that require this line.

File: app/schemas/general_schema.rb (hunk 4 above)
Comment (mcritchlow) [reply app/schemas/general_schema.rb:47]: Gotcha. I'm just wondering why leave the line in there if we don't need it?

File: config/initializers/hydra_config.rb (hunk 5 above)
Comment (mcritchlow) [reply config/initializers/hydra_config.rb:23]: Good to know, thanks!

File: app/schemas/general_schema.rb (hunk 4 above)
Comment (lsitu) [reply app/schemas/general_schema.rb:47]: Yep, good question. I was thinking just about leaving it as a marker here for now and we can clean it up later with others predicates that were commented out earlier once we finalize all the predicates. But yes, we can do it now if you prefer.

File: app/actors/hyrax/actors/environment.rb (hunk 2 above)
Comment (lsitu) [reply app/actors/hyrax/actors/environment.rb:3]: Right, this was introduced a couple of months earlier. It was updated recently for rails 5.1, which breaks our blank nodes support. I can't find a better way to fix that so it's customized for our need.

**Comment by mcritchlow** [comment]
@hweng - any other thoughts/concerns?

**Comment by hweng** [comment]
LGTM :+1:

**Pull request merged**